include/pybind11/cast.h

```diff
@@ -1890,6 +1890,14 @@ template <return_value_policy policy = return_value_policy::automatic_reference,
     return result;
 }
 
+#if PY_MAJOR_VERSION >= 3
+template <typename... Args,
+          typename = detail::enable_if_t<args_are_all_keyword_or_ds<Args...>()>>
+object make_namespace(Args&&... args_) {
+    return reinterpret_steal<object>(_PyNamespace_New(dict(std::forward<Args>(args_)...).ptr()));
```

Comment:
Can `_PyNamespace_New` return `NULL` and set an error inside CPython?

Reply:
https://github.com/python/cpython/blob/63298930fb531ba2bb4f23bc3b915dbf1e17e9e1/Objects/namespaceobject.c#L33

It can return NULL, but not sure about setting the error. Error part is on `PyDict_New`.

Reply:
It can indeed return NULL. I'm not sure this will actually happen, but I revised it anyway so that it is more explicit.